### PR TITLE
Regenesis of `SpentMessages` and `ProcessedTransactions`

### PR DESCRIPTION
**Pull request by xgreenx**
Re-implementation of the https://github.com/FuelLabs/fuel-core/pull/1829
Work towards https://github.com/FuelLabs/fuel-core/issues/1790

The change migrates the `ProcessedTransactions` table as part of the on-chain snapshot. Since it is used to validate blocks, the root of all processed transactions must be part of the `Genesis` structure along with messages, coins, and contracts.

The changes also moved `SpentMessages` from the on-chain database to the off-chain database since now it is only used for the GraphQL API.

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

**Review by Dentosal — DISMISSED** [review]
Looks good otherwise, but maybe we should have some test verifying the behavior?

**Review by Dentosal — DISMISSED** [review]

File: crates/services/consensus_module/poa/src/service.rs [hunk 1]
```diff
@@ -459,26 +459,37 @@ where
...
-                        self.last_block_created = last_block_created;
-                    }
+                _ = state.changed() => {
+                    break;
+                }
+                _ = self.tx_status_update_stream.next() => {
+                    // ignore txpool events while syncing
+                }
+                _ = self.timer.wait() => {
+                    // ignore timer events while syncing
```
Comment (xgreenx) [inline crates/services/consensus_module/poa/src/service.rs:478]: During debugging tests, I realized that for `Never` mode, we will always push events but not process them, causing a huge buffer. Just add ignoring of them here to remove the from the buffer.

File: bin/fuel-core/chainspec/testnet/state_transition_bytecode.wasm
Comment (Dentosal) [inline bin/fuel-core/chainspec/testnet/state_transition_bytecode.wasm:1]: Why are these removed?

**Review by Dentosal — APPROVED** [review]

File: bin/fuel-core/chainspec/testnet/state_transition_bytecode.wasm
Comment (xgreenx) [reply bin/fuel-core/chainspec/testnet/state_transition_bytecode.wasm:1]: Only removed 895 bytes=D  The file still exists
image
